### PR TITLE
don't pre-bind pods to nodes and don't create node objects

### PR DESCRIPTION
**Pull request by tzneal**
**1. Issue, if available:**
Multiple

**2. Description of changes:**

- Eliminate binding of pods to nodes
- Elminate node creation
- add a command line flag to enable the pod-eni resource
- remove the Karpenter specific not-ready taint

**3. How was this change tested?**

Unit testing & deploying to EKS

**4. Does this change impact docs?**
- [X] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by trondhindenes** [comment]
I've tried googling around for "pod binding" but I can't find a definitive answer. Could you briefly explain what is meant by it? I came here from https://github.com/aws/karpenter/issues/1459

**Comment by olemarkus** [comment]
See https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/

Pod binding is the actual assigning of Pods to Nodes. The way Karpenter works right now, it tries to predict what the final Node object will look like, creates the actual resource, and then binds Pods to it. All this before the instance has even been provisioned. Unfortunately, Karpenter in many cases, like yours, cannot know exactly what the Node object will look like in the end when all controllers have done their thing to it. Or because kubelet resets parts of the Node object when it starts up, such as the GPU count.

Also because Karpenter doesn't implement the full scheduler, it leads to different scheduling rules during capacity increase than during say a deployment rollout.

This PR aims to remove the Node creation and let kubelet create it, and remove the Pod binding and let kube-scheduler handle that part.

Not that brief, but hope it clarifies :)

**Comment by trondhindenes** [comment]
@olemarkus you're everywhere :-)

File: website/content/en/preview/upgrade-guide/_index.md [hunk 1]
```diff
@@ -69,6 +69,11 @@ for a subset of older versions and deprecate the others.
 
 # Released Upgrade Notes
 
+## Upgrading to v0.11.0+
+
+v0.11.0 changes the way that the `vpc.amazonaws.com/pod-eni` resource is reported.  Instead of being reported for all nodes that could support the resources regardless of if the cluster is configured to support it, it is now controlled by a command line flag or environment variable. The parameter defaults to false and must be set if your cluster uses [security groups for pods](https://docs.aws.amazon.com/eks/latest/userguide/security-groups-for-pods.html).  This can be enabled by setting the environment variable `AWS_ENABLE_POD_ENI` to true via the helm value `controller.env`. 
+
+
```
Comment (njtran) [inline website/content/en/preview/upgrade-guide/_index.md:76]: This should be in the v0.11.0 release, not here. Right?

**Comment by tzneal** [comment]
Closing this for now while we rethink how to not create nodes, but not regress with respect to node visibility during startup.

**Pull request closed without merging**